acoustid/ratelimiter.py

```
@@ -19,6 +20,17 @@ def __init__(self, redis, prefix, interval=20, steps=4):
         self.steps = steps
 
+    def seconds_until_next_step(self):
+        # type: () -> int
+        """Seconds until the oldest step of the sliding window expires.
+
+        The window is split into `steps` buckets keyed on absolute time, so
+        capacity frees up when the current bucket rolls over. Rounded up, and
+        never zero, so it can be used as a Retry-After value.
+        """
+        step = self.interval / self.steps
+        return max(1, int(math.ceil(step - (time.time() % step))))
```

Ruff (0.16.0)

[warning] Value being cast to `int` is already an integer

Remove unnecessary `int` call

(RUF046)